packages-exp/auth-exp/src/core/mfa/mfa_user.test.ts

```diff
@@ -0,0 +1,234 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import { testAuth, testUser } from '../../../test/mock_auth';
+import * as mockFetch from '../../../test/mock_fetch';
+import { Auth } from '../../model/auth';
+import { MultiFactorSessionType } from './mfa_session';
+import { MultiFactorUser } from './mfa_user';
+import { PhoneAuthProvider } from '../providers/phone';
+import {
+  PhoneMultiFactorGenerator,
+  PhoneMultiFactorAssertion
+} from './assertions/phone';
+import { mockEndpoint } from '../../../test/api/helper';
+import { Endpoint } from '../../api';
+import { IdTokenResponse } from '../../model/id_token';
+import { APIUserInfo } from '../../api/account_management/account';
+import { MultiFactorInfo } from './mfa_info';
+import { ServerError } from '../../api/errors';
+
+use(chaiAsPromised);
+
+describe('core/mfa/mfa_user', () => {
+  let auth: Auth;
+  let mfaUser: MultiFactorUser;
+
+  beforeEach(async () => {
+    auth = await testAuth();
+    mockFetch.setUp();
+    mfaUser = new MultiFactorUser(testUser(auth, 'uid', undefined, true));
+  });
+
+  afterEach(mockFetch.tearDown);
+
+  describe('getSession', () => {
+    it('should return the id token', async () => {
+      const mfaSession = await mfaUser.getSession();
+      expect(mfaSession.type).to.eq(MultiFactorSessionType.ENROLL);
+      expect(mfaSession.credential).to.eq('access-token');
+    });
+  });
+
+  describe('enroll', () => {
+    let finalizeMfaEnrollmentMock: mockFetch.Route;
+    let assertion: PhoneMultiFactorAssertion;
+
+    const serverUser: APIUserInfo = {
```

Comment:
we use this a lot, we really ought to throw it in a constant in a test helper somewhere

Reply:
will do in follow up PR, this one is already unwieldy